sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -152,7 +168,12 @@ object NestedColumnAliasing {
     val exclusiveAttrSet = AttributeSet(exclusiveAttrs ++ otherRootReferences)
     val aliasSub = nestedFieldReferences.asInstanceOf[Seq[ExtractValue]]
       .filter(!_.references.subsetOf(exclusiveAttrSet))
-      .groupBy(_.references.head)
+      .groupBy(t => (t.references.head.exprId, t.references.head.dataType))
+      // The above groupBy is to avoid situation like the following situation.
+      // For example, `exprIdA -> List(a, b)` and  `exprIdA -> List(c, d)`
```

Comment:
I'm not sure about what `exprIdA -> List(a, b)` means? I think we need to explain that attribute might have different qualifiers so we cannot well group up semantically equal attributes here.

Reply:
Just pushed another comment.